### PR TITLE
Add documentation snippet for using partials

### PR DESCRIPTION
**Pull request by santoso-wijaya**
- [N/A] Tests created for any new feature or regression tests for bugfixes.

#509

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -92,3 +92,35 @@ See
...
+as template partials.
+
+Example:
+
+```liquid
+# common.liquid
+Number: {{ i }}
+```
+
+```rust
```
Comment (epage) [inline README.md:108]: imo this should be a module level comment on `partials`.
- Focuses the documentation in one place
- Verifies the examples

Comment (santoso-wijaya) [reply README.md]: Thanks for the quick review! I moved the documentation as a mod rustdoc as you suggested.

File: crates/core/src/partials/mod.rs [hunk 2]
```diff
@@ -1,3 +1,36 @@
...
+//!
+//! ```liquid
+//! # common.liquid
+//! Number: {{ i }}
+//! ```
+//!
+//! ```rust,no_run,compile_fail
+//! // Build template partials using an eager, in-memory source compiler.
+//! // Other compilation policies also exist depending on specific needs.
+//! type Partials = EagerCompiler<InMemorySource>;
```
Comment (santoso-wijaya) [inline crates/core/src/partials/mod.rs:15]: These types (along with ParserBuilder, Parser, etc) are defined outside of this crate in a reverse dependent way, and I can't figure out what the right `# use ...` directive to use to make this snippet compile.

File: crates/core/src/partials/mod.rs [hunk 3]
```diff
@@ -1,3 +1,36 @@
...
+//!
+//! let partials = {
+//!   let mut _partials = Partials::empty();
+//!
+//!   let filepath = String::from("common.liquid");
+//!   let contents = std::fs::read_to_string(filepath).unwrap();
```
Comment (santoso-wijaya) [inline crates/core/src/partials/mod.rs:21]: What to do with file reads in doctests

File: crates/core/src/partials/mod.rs [hunk 4]
```diff
@@ -1,3 +1,36 @@
...
+//! Number: {{ i }}
+//! ```
+//!
+//! ```rust,no_run,compile_fail
+//! // Build template partials using an eager, in-memory source compiler.
+//! // Other compilation policies also exist depending on specific needs.
+//! type Partials = EagerCompiler<InMemorySource>;
+//!
+//! let partials = {
+//!   let mut _partials = Partials::empty();
```
Comment (epage) [inline crates/core/src/partials/mod.rs:18]: Could you remove the `_` prefix?  That tells the compiler that it can be unused.  We can just call it `partials`

Comment (santoso-wijaya) [reply crates/core/src/partials/mod.rs]: Done.

File: crates/core/src/partials/mod.rs [hunk 5]
```diff
@@ -1,3 +1,36 @@
...
+//! included files as template partials.
+//!
+//! Example:
+//!
+//! ```liquid
+//! # common.liquid
+//! Number: {{ i }}
+//! ```
+//!
+//! ```rust,no_run,compile_fail
```
Comment (epage) [inline crates/core/src/partials/mod.rs:12]: Sorry, I had overlooked your comments that you had resolved.

I'm fine with `no_run` examples but I really dislike `ignore` (which is more of what this should be than `compile_fail`).

Instead, let's stop re-exporting `partials` in `liquid` and instead have a dedicated module that manually re-exports the types from `liquid_core::partials`.

What would be helpful is if we organized the PR as the following commits
- A commit to only re-export `partials` types and not the `mod` itself
- A commit that adds the documentation.

As those are the two core atomic operations for this PR.

Comment (santoso-wijaya) [reply crates/core/src/partials/mod.rs:12]: Got it.

There is still an issue with `ParserBuilder` being defined in in a layer above `liquid_core::partials` here, though.

So, in the rustdoc code that we include here in `crates/core/src/partials/mod.rs`, we would be showing how a user would create a partial to be included in a `liquid::ParserBuilder`, but `liquid::ParserBuilder` being a client of `liquid_core::partials` will not be in scope here to be imported in the rustdoc code.

In other words, this compiles just fine:

```rust

//! ```rust,no_run
//! use liquid_core::partials::{EagerCompiler, InMemorySource, PartialSource};

```

But the following will not:

```rust

//! ```rust,no_run
//! use liquid_core::partials::{EagerCompiler, InMemorySource, PartialSource};
//! use liquid::ParserBuilder;

```

Comment (epage) [reply crates/core/src/partials/mod.rs]: No, I was saying we should create `src/partials.rs` and have the example on that.

Comment (santoso-wijaya) [reply crates/core/src/partials/mod.rs]: Creating `src/partials.rs` makes sense to me. The rustdoc example that we can write in that module won't be able to include and compile this usage snippet though:

```rust
//! // Compile and render the main template, which uses the "common" partial.
//! let parser = ParserBuilder::with_stdlib().partials(partials).build().unwrap();
```

Since `liquid::ParserBuilder` depends on `partials`, not the other way around.

Comment (santoso-wijaya) [reply crates/core/src/partials/mod.rs]: Wait, nevermind, I think that'll work. Lemme prep the commits.

Comment (santoso-wijaya) [reply crates/core/src/partials/mod.rs]: Wait, nevermind, I think that'll work. I'll prep the commits.

Comment (santoso-wijaya) [reply crates/core/src/partials/mod.rs]: Done.

**Comment by epage** [comment]
Thank you for your work and our patience on the feedback!
